src/impl/results_notifier.cpp

```diff
@@ -149,20 +158,31 @@ void ResultsNotifier::calculate_changes()
         m_changes = CollectionChangeBuilder::calculate(m_previous_rows, next_rows,
                                                        get_modification_checker(*m_info, *m_query->get_table()),
                                                        move_candidates);
-
         m_previous_rows = std::move(next_rows);
+
+#if REALM_ENABLE_SYNC
+        if (m_partial_sync_enabled)
+            m_previous_partial_sync_status_code = partial_sync::create_or_update_subscription(*m_sg, m_changes, *m_query, m_previous_partial_sync_status_code);
```

Comment:
ResultsNotifier should not be managing partial sync subscriptions. That's pretty unrelated to calculating changesets for Results.